src/emu/diimage.h

```diff
@@ -144,13 +142,11 @@ class device_image_interface : public device_interface
 	static const char *device_brieftypename(iodevice_t type);
 	static iodevice_t device_typeid(const char *name);
 
-	virtual void device_compute_hash(util::hash_collection &hashes, const void *data, size_t length, const char *types) const;
-
 	virtual image_init_result call_load() { return image_init_result::PASS; }
 	virtual image_init_result call_create(int format_type, util::option_resolution *format_options) { return image_init_result::PASS; }
 	virtual void call_unload() { }
 	virtual std::string call_display() { return std::string(); }
-	virtual device_image_partialhash_func get_partial_hash() const { return nullptr; }
+	virtual uint32_t unhashed_header_length() const { return 0; }
```

Comment:
`src/emu` is supposed to use the abbreviated types for consistence - that applies to headers as well as sources